### PR TITLE
#455 programs router supports retrieval and updating of participant activities completion status

### LINKED ISSUE #455: programsRouter should support retrieval and updating of ParticipantActivities completion status
## Expected behaviour

There should be two programs router functions that return a participant activity's `completed` status and allow for the value of that attribute to be modified.

## Details and resources

This ticket is associated with #454 and #457 in order to fulfill the product spec requirement for a [user to be able to mark an activity as complete](https://docs.google.com/document/d/1bYpceHQJZmm8h4opFTvTvZua1Xn39E1ZfNYDiDds7PM/edit#heading=h.q7slb8flu34l).

The two `programsRouter` functions should be something along the lines of:
- `programsRouter.get('/activityStatus/:programId/:activityId')` - returns the `completed` status of a given `participantActivity`. this function will allow a user to get their own status (admin access to be addressed later on)
- `programsRouter.post('/activityStatus/:programId/:activityId')` - modifies the `completed` value of an existing `participantActivity` (`POST` instead of `PUT` since we're creating new records instead of updating them)
  - in the POST body, there should be info about what the `completed` value is being changed to (eg, `{"completed": true}`)
  - ~~associated service function should only allow activities of `type: assignment` to have this attribute updated~~

The functions should take the following parameters: `principal_id` (corresponding to the user whose activity status is being tracked), `program_id`, and `activity_id`

`__tests__/programsRouter.ts` should also be updated to test these new router functions.

## Checklist

- [x] This issue is linked to the appropriate project.
- [x] This issue has appropriate labels.

### PR DESCRIPTION
**Pull request by rrabinovitch**
## Proposed changes

This PR resolves #455 by creating getter and setter functions for individual an participantActivity's completion status, as well as tests for these new functions.

Manually tested on local machine:
- GET `http://api.rhi.zone-development/programs/activityStatus/1/1` returned `{"data":{"programId":1,"activityId":1,"completed":false}}`
- POST `http://api.rhi.zone-development/programs/activityStatus/1/1` with request body `{"completed": true}` returned `{"data":{"completed":false}}`

## Checklist

- [x] Are the issues being addressed linked to this PR?
- [x] Do all commit messages start with the issue number? - a few merges from main are missing this
- [x] Are all code changes sufficiently tested?

### CODE REVIEW

File: api/src/middleware/programsRouter.ts [hunk 1]
```diff
@@ -18,4 +23,84 @@ programsRouter.get('/', async (req, res, next) => {
...
+    if (!Number.isInteger(programIdNum) || programIdNum < 1) {
+      next(new BadRequestError(`"${programIdNum}" is not a valid program id.`));
+      return;
+    }
+
+    if (!Number.isInteger(activityIdNum) || activityIdNum < 1) {
+      next(new BadRequestError(`"${activityId}" is not a valid activity id.`));
+      return;
+    }
+
```
Comment (veranika-karpava) [inline api/src/middleware/programsRouter.ts:44]: I noticed that we didn't check that programId and activityId are existing.

File: api/src/middleware/programsRouter.ts [hunk 2]
```diff
@@ -18,4 +23,84 @@ programsRouter.get('/', async (req, res, next) => {
...
+        activityIdNum,
+        completed
+      );
+    } catch (error) {
+      next(error);
+      return;
+    }
+
+    res.status(201).json(itemEnvelope(updatedParticipantActivity));
+  }
```
Comment (veranika-karpava) [inline api/src/middleware/programsRouter.ts:103]: Also Phoebe updated this service function.
I'm not sure did she update her branch or main branch.

**Review by veranika-karpava — CHANGES_REQUESTED** [review]
During the testing I got data from get and put request. 
Great job, Ruthie!

**Review by seidior — APPROVED** [review]
Like I mentioned before, thank you for all the work on this ticket, and thank you for being adaptable to all of the changes in approach and style. I'm going to force-merge this PR tonight since it's a roadblock for the frontend folks, but it's so late in the day already after we fixed the service function issues, it's too late to ask you to merge it :)

**Pull request merged**